### ISSUE TITLE
[Feature Request]: Fetch trending books from the books api

### ISSUE THREAD

**Issue by aasiflm10** [Trisha-tech/OnlineBookSales#353]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

Right now the trending and new arrivals shows : 
![image](https://github.com/user-attachments/assets/52904ffe-161d-458a-9bd3-459df7ba7fab)

We can update it to add real books using some api.

### Use Case

Account for better UI UX and usability of the product

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

**Comment by Trisha-tech** [comment]
> ### Is there an existing issue for this?
> * [x]  I have searched the existing issues
> 
> ### Feature Description
> Right now the trending and new arrivals shows : ![image](https://private-user-images.githubusercontent.com/94681493/374245008-52904ffe-161d-458a-9bd3-459df7ba7fab.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgzODAxOTAsIm5iZiI6MTcyODM3OTg5MCwicGF0aCI6Ii85NDY4MTQ5My8zNzQyNDUwMDgtNTI5MDRmZmUtMTYxZC00NThhLTliZDMtNDU5ZGY3YmE3ZmFiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA4VDA5MzEzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ4NjlkZDZlYzgzMTY0MmJhYjMwNjJjZmI4OTc1ZTMzZDI5ZGUyMDQwOTgzN2M0Y2IzOTQ4M2NhNGVlNjI2OGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.XzGjMOQWol4UIncfPyXCy2N9v-5IYu7DoytaWmYL2wU)
> 
> We can update it to add real books using some api.
> 
> ### Use Case
> Account for better UI UX and usability of the product
> 
> ### Benefits
> _No response_
> 
> ### Add ScreenShots
> _No response_
> 
> ### Priority
> High
> 
> ### Record
> * [x]  I have read the Contributing Guidelines
> * [x]  I'm a GSSOC'24 contributor
> * [x]  I have starred the repository

Hello @aasiflm10, Go for it
All the Best

**Comment by aasiflm10** [comment]
thx. working on it.

**Comment by aasiflm10** [comment]
@Trisha-tech i am getting these errors :
![image](https://github.com/user-attachments/assets/ff42d4b6-2cd5-472b-9ec9-2407031687c0)

I have already done npm install in both the root directory and the client directory. Please tell if i am missing something.

**Comment by aasiflm10** [comment]
There is no index.js file in Components. Need to add a index.js to /client/src/Components to resolve this error. 
Also i am not able to sign up or Log in. Do you have any test user credentials to access the application?

**Comment by Trisha-tech** [comment]
> There is no index.js file in Components. Need to add a index.js to /client/src/Components to resolve this error. Also i am not able to sign up or Log in. Do you have any test user credentials to access the application?

![Screenshot_2024-10-08-17-17-12-940_com android chrome](https://github.com/user-attachments/assets/79110cb2-eed3-4f4d-b2e6-df960c15acde)
By giving the correct path of Navbar, Footer, Profile Page, Product will solve the issue. 
Register and Signin functionality is under progress.

**Comment by aasiflm10** [comment]
@Trisha-tech  i tried adding the correct path but again it falls back to "there was no index.js file in Components". 
I built one and the project works fine.

Fetching trending books from the books api also works fine : 
![image](https://github.com/user-attachments/assets/cf5f5fb0-3fd5-4f26-93e2-6f0134e20697)

Here is a video that shows the working and fetch code. 
https://drive.google.com/file/d/1cDcA_d6JISoj_b6kBQVuhw2-ZxdZmmhx/view?usp=sharing

I have used Newyork Times Books API. So we will have to add NY_TIMES_API_KEY. I tried adding it to .env but the client directory is not able to load env variables. I tried finding the issue but could load the env variables to client. 
(NOTE : env variables load find in the backend server ).

Shall i move forward with a PR?
